pkg/sql/insert_fast_path.go

```diff
@@ -282,6 +283,14 @@ func (n *insertFastPathNode) BatchedNext(params runParams) (bool, error) {
 			var err error
 			inputRow[col], err = typedExpr.Eval(params.EvalContext())
 			if err != nil {
+				// Intercept parse error due to ALTER COLUMN TYPE schema change.
+				if n.run.insertRun.insertCols[col].AlterColumnTypeDropColumn {
+					code := pgerror.GetPGCode(err)
+					if code == "22P02" {
+						return false, errors.Wrapf(err,
```

Comment:
I'd make this error a bit more general -- something like "this column is still finalizing the alter column set datatype schema change ..."